src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1436,37 +1422,24 @@ void SolverInterfaceImpl::writeBlockVectorGradientData(
     auto &     gradientValuesInternal = data.gradientValues();
     const auto vertexCount            = gradientValuesInternal.cols() / data.getDimensions();
 
+    // gradient matrices input one after the other (read row-wise)
     for (int i = 0; i < size; i++) {
```

Comment:
This can also be simplified with block operations:

```cpp
    Eigen::Map<const Eigen::MatrixXd> gradients(gradientValues, _dimensions, _dimensions * size);

    // gradient matrices input one after the other (read row-wise)
    for (int i = 0; i < size; i++) {
      const auto valueIndex = valueIndices[i];
      PRECICE_CHECK(0 <= valueIndex && valueIndex < vertexCount,
                    "Cannot write gradient data \"{}\" to invalid Vertex ID ({}). Please make sure you only use the results from calls to setMeshVertex/Vertices().",
                    data.getName(), valueIndex);

      data.gradientValues().block(0, _dimensions*valueIndex, _dimensions, _dimensions)
                = gradients.block(0, _dimensions*i,          _dimensions, _dimensions);
    }
```

Reply:
This version is very nice, I overlooked it and will apply it to the other function as well.